src/pages/Room/Room.tsx

```diff
@@ -113,20 +114,28 @@ const Dashboard = () => {
                 className={clsx(style.splitter, style["splitter-verticle"])}
                 onStopResize={() => resetEditorLayout()}
               />
-              <ReflexElement flex={0.45}>
-                <ReflexContainer orientation="horizontal">
-                  <ReflexElement style={{ display: "flex" }}>
-                    <CodeMirror editorInstance={editorInstance} setEditorInstance={setEditorInstance} />
-                  </ReflexElement>
-                  <ReflexSplitter
-                    className={clsx(style.splitter, style["splitter-horizontal"])}
-                    onStopResize={() => resetEditorLayout()}
-                  />
-                  <ReflexElement flex={0.3}>
-                    <InputOutputFile rows={rows} TextAreaRef={TextAreaRef} editorInstance={editorInstance} />
-                  </ReflexElement>
-                </ReflexContainer>
-              </ReflexElement>
+              {/*this context provider share the code execution information between CodeMirror and Input Output File,
+                this help you to execute the code from inside of each component*/}
+              <CodeExecutionInfoProvider>
+                <ReflexElement flex={0.45}>
+                  <ReflexContainer orientation="horizontal">
+                    <ReflexElement style={{ display: "flex" }}>
+                      <CodeMirror editorInstance={editorInstance} setEditorInstance={setEditorInstance} />
+                    </ReflexElement>
+                    <ReflexSplitter
+                      className={clsx(style.splitter, style["splitter-horizontal"])}
+                      onStopResize={() => resetEditorLayout()}
+                    />
+                    <ReflexElement flex={0.3}>
+                      <InputOutputFile
+                        rows={rows}
+                        TextAreaRef={TextAreaRef}
+                        editorInstance={editorInstance}
+                      />
+                    </ReflexElement>
+                  </ReflexContainer>
+                </ReflexElement>
+              </CodeExecutionInfoProvider>
```

Comment:
Can you add this CodeExecutionProvider at the top level, i.e. [`App.tsx`](https://github.com/Rishabh-malhotraa/caucus/blob/main/src/App.tsx) where all the other context is defined.

Reply:
It is because of performance, In Context If you wanna get the best performance, you should place the provider as close to the files as it needed.

Reply:
Okay, It should make that much difference. Defining it in `App.tsx` is just two parent nodes up the hierarchy.

Reply:
Should i refactor the code as it was ? I mean, last night i did requested changes

Reply:
No no, the new updates are fine!